config/v1/types_scheduling.go

```diff
@@ -22,12 +22,16 @@ type Scheduler struct {
 }
 
 type SchedulerSpec struct {
-	// policy is a reference to a ConfigMap containing scheduler policy which has
+	// DEPRECATED: policy is a reference to a ConfigMap containing scheduler policy which has
 	// user specified predicates and priorities. If this ConfigMap is not available
 	// scheduler will default to use DefaultAlgorithmProvider.
 	// The namespace for this configmap is openshift-config.
 	// +optional
-	Policy ConfigMapNameReference `json:"policy"`
+	Policy ConfigMapNameReference `json:"policy,omitempty"`
+	// profiles is a reference to a ConfigMap containing a list of custom scheduler profiles
+	// Note that this config will eventually replace the Policy config currently available
+	// +optional
+	Profiles ConfigMapNameReference `json:"profiles"`
```

Comment:
IIRC our recent discussion with @deads2k it was meant to look more like audit profiles: https://github.com/openshift/api/blob/master/config/v1/types_apiserver.go#L58

Reply:
We could do something like that. Looking at [the enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/audit-policy.md) I think a couple defined "presets" of scheduler profiles could help bridge from what we currently have. If it isn't enough for some users, we can always revisit exposing more of the config.

I'll work on a similar enhancement to the audit profiles one to define a couple scheduling profiles that will hopefully cover most use cases.

Reply:
> If it isn't enough for some users, we can always revisit exposing more of the config.

We'll have the deprecated policy handy with metrics attached to it. We'll use that as fallback and learn how many users actually uses that. We can always expand these profiles based on feedback.